### PR TITLE
cmusfm: use versiont tags, add compile options

### PR DESCRIPTION
**Pull request by p-h**
###### Motivation for this change
Use version numbers if possible. I also wanted to be able to easily turn on debugging.

###### Things done
Use tags instead of commit hashes, make libnotify and debug compile options overridable

```
for hash in x27w90b06bgi3mpnb0608bk4kbif6kjz 5hnll6ym04gqgaqa4syg9grkw5cyxbfh c44zk08q26xv6z52x7xbi2f0gljmq40b ylwlpm0ipggq826q8i9jnac2drpq69ni; do /nix/store/$hash-cmusfm-0.4.1/bin/cmusfm; done
usage: /nix/store/x27w90b06bgi3mpnb0608bk4kbif6kjz-cmusfm-0.4.1/bin/cmusfm [init]

NOTE: Before usage with the cmus you should invoke this program with the
      `init` argument. Afterwards you can set the status_display_program
      (for more information see `man cmus`). Enjoy!
usage: /nix/store/5hnll6ym04gqgaqa4syg9grkw5cyxbfh-cmusfm-0.4.1/bin/cmusfm [init]

NOTE: Before usage with the cmus you should invoke this program with the
      `init` argument. Afterwards you can set the status_display_program
      (for more information see `man cmus`). Enjoy!
usage: /nix/store/c44zk08q26xv6z52x7xbi2f0gljmq40b-cmusfm-0.4.1/bin/cmusfm [init]

NOTE: Before usage with the cmus you should invoke this program with the
      `init` argument. Afterwards you can set the status_display_program
      (for more information see `man cmus`). Enjoy!
usage: /nix/store/ylwlpm0ipggq826q8i9jnac2drpq69ni-cmusfm-0.4.1/bin/cmusfm [init]

NOTE: Before usage with the cmus you should invoke this program with the
      `init` argument. Afterwards you can set the status_display_program
      (for more information see `man cmus`). Enjoy!
```

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by CharlesHD — APPROVED** [review]
Hey cool, it's a lot cleaner and configurable. Thank you for your contribution !

File: pkgs/applications/audio/cmusfm/default.nix [hunk 1]
```diff
@@ -1,21 +1,23 @@
-{ lib, stdenv, fetchFromGitHub, autoreconfHook, pkg-config, curl, libnotify, gdk-pixbuf }:
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, pkg-config, curl, libnotify
+, gdk-pixbuf, libnotifySupport ? true, debug ? false }:
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/audio/cmusfm/default.nix:2]:
```suggestion
, gdk-pixbuf, libnotifySupport ? stdenv.isLinux, debug ? false }:
```

Maybe is not a very strong argument but at least it should be disabled on darwin.

**Review by CharlesHD — APPROVED** [review]

**Comment by p-h** [comment]
bump

File: pkgs/applications/audio/cmusfm/default.nix [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, pkg-config, curl, libnotify
+, gdk-pixbuf, libnotifySupport ? stdenv.isLinux, debug ? false }:
 
-stdenv.mkDerivation {
-  version = "2021-05-19";
-  pname = "cmusfm-unstable";
+stdenv.mkDerivation rec {
+  pname = "cmusfm";
+  version = "0.4.1";
   src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/cmusfm/default.nix:7]:
```suggestion
  version = "0.4.1";

  src = fetchFromGitHub {
```

File: pkgs/applications/audio/cmusfm/default.nix [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
     owner = "Arkq";
-    repo = "cmusfm";
-    rev = "a1f9f37c5819ca8a5b48e6185c2ec7ad478b9f1a";
-    sha256 = "19akgvh9gl99xvpmzgqv88w2mnnln7k6290dr5rn3h6a1ihvllaw";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1px2is80jdxchg8cpn5cizg6jvcbzyxl0qzs3bn0k3d10qjvdww5";
   };
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/cmusfm/default.nix:12]:
```suggestion
  };

```

File: pkgs/applications/audio/cmusfm/default.nix [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
-  # building
-  configureFlags = [ "--enable-libnotify" ];
+  configureFlags = lib.optional libnotifySupport "--enable-libnotify"
+    ++ lib.optional debug "--enable-debug";
   nativeBuildInputs = [ autoreconfHook pkg-config ];
-  buildInputs = [ curl libnotify gdk-pixbuf ];
+  buildInputs = [ curl gdk-pixbuf ] ++ lib.optional libnotifySupport libnotify;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/cmusfm/default.nix:16]:
```suggestion
  buildInputs = [ curl gdk-pixbuf ]
    ++ lib.optional libnotifySupport libnotify;
```

File: pkgs/applications/audio/cmusfm/default.nix [hunk 5]
```diff
@@ -1,18 +1,22 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1px2is80jdxchg8cpn5cizg6jvcbzyxl0qzs3bn0k3d10qjvdww5";
   };
-  # building
-  configureFlags = [ "--enable-libnotify" ];
+
+  configureFlags = lib.optional libnotifySupport "--enable-libnotify"
+    ++ lib.optional debug "--enable-debug";
   nativeBuildInputs = [ autoreconfHook pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/cmusfm/default.nix:17]:
```suggestion
    ++ lib.optional debug "--enable-debug";

  nativeBuildInputs = [ autoreconfHook pkg-config ];

```

**Pull request merged**